templates/datadog-agent.yaml

```diff
@@ -0,0 +1,43 @@
+{{ if .Values.datadogAgent.enabled }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: datadog-agent
+  labels:
+    release: datadog-agent
+    istio-injection: disabled
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: datadog-api-key
+  namespace: datadog-agent
+stringData:
```

Comment:
Interesting, I am not sure if stringData works well in helm chart. You can also use base64 enc function in helm.